manifests/vanilla/csi-snapshot-validatingwebhook.yaml

```diff
@@ -0,0 +1,70 @@
+# Requires k8s 1.20+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: snapshot-validation-service
+  namespace: kube-system
+spec:
+  selector:
+    app: snapshot-validation
+  ports:
+    - protocol: TCP
+      port: 443 # Change if needed
+      targetPort: 443 # Change if the webserver image expects a different port
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: "validation-webhook.snapshot.storage.k8s.io"
+webhooks:
+  - name: "validation-webhook.snapshot.storage.k8s.io"
+    rules:
+      - apiGroups:   ["snapshot.storage.k8s.io"]
+        apiVersions: ["v1"]
+        operations:  ["CREATE", "UPDATE"]
+        resources:   ["volumesnapshots", "volumesnapshotcontents"]
+        scope:       "*"
+    clientConfig:
+      service:
+        namespace: kube-system
+        name: "snapshot-validation-service"
+        path: "/volumesnapshot"
+      caBundle: ${CA_BUNDLE}
+    admissionReviewVersions: ["v1"]
+    sideEffects: None
+    failurePolicy: Ignore # We recommend switching to Fail only after successful installation of the webhook server and webhook.
+    timeoutSeconds: 2 # This will affect the latency and performance. Finetune this value based on your application's tolerance.
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: snapshot-validation-deployment
+  namespace: kube-system
+  labels:
+    app: snapshot-validation
+spec:
+  replicas: 3
```

Comment:
Should this be 3 always ? Or only in case of multi-master setups ?. If so, can you clarify with comments

Reply:
The recommendation is 3 as per https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/webhook-example/webhook.yaml, so from my understanding, this seems to be standard irrespective of whether it's a HA setup.